packages/near-membrane-base/src/membrane.ts

```diff
@@ -661,6 +662,17 @@ export function createMembraneMarshall() {
             return desc;
         }
 
+        function isPlainObjectOrEquivalent(targetPointer: Pointer): boolean {
```

Comment:
the pointer is for objects only, arrays are treated before calling this, so it is safe to just go with the proto analysis directly.

Reply:
This should become a callback so all these operations are happening locally in the blue side so it will be faster.